plugin-server/src/main/ingestion-queues/session-recording/session-recordings-blob-consumer.ts

```diff
@@ -348,7 +352,17 @@ export class SessionRecordingBlobIngester {
             this.sessions.forEach((sessionManager) => {
                 sessionManangerBufferSizes += sessionManager.buffer.size
 
-                void sessionManager.flushIfSessionBufferIsOld().catch((err) => {
+                const flushTolerance = this.serverConfig.SESSION_RECORDING_MAX_BUFFER_AGE_SECONDS * 1000
+                // in practice, we will always have a values for latestKaftaMessageTimestamp,
+                // but in case we get here before the first message, we use now
+                const referenceNow = this.latestKafkaMessageTimestamp || DateTime.now().toMillis()
+                const thirtyMinutesInMilliseconds = 30 * 60 * 1000
+                const flushThresholdMillis =
+                    DateTime.now().toMillis() - referenceNow > thirtyMinutesInMilliseconds
+                        ? flushTolerance * 3
+                        : flushTolerance
```

Comment:
Nice that make sense! I guess we could do this later as a config option like `SESSION_RECORDING_MAX_BUFFER_AGE_EXTENSION_SECONDS` or something.